iModelJsNodeAddon/JsInterop.cpp

```diff
@@ -922,6 +923,70 @@ DbResult JsInterop::ImportSchemas(DgnDbR dgndb, bvector<Utf8String> const& schem
     return dgndb.SaveChanges();
     }
 
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+DbResult JsInterop::ConvertEC2XmlSchemas(DgnDbR dgndb, bvector<Utf8String> const& ec2XmlStrings, bvector<Utf8String>& ec3XmlStrings)
+    {
+    if (0 == ec2XmlStrings.size())
+        return BE_SQLITE_ERROR;
+
+    ECSchemaReadContextPtr schemaContext = ECSchemaReadContext::CreateContext(false /*=acceptLegacyImperfectLatestCompatibleMatch*/, true /*=includeFilesWithNoVerExt*/);
+    JsInterop::AddFallbackSchemaLocaters(dgndb, schemaContext);
+
+    bvector<bpair<SchemaKey, ECSchemaPtr>> schemaKeyPairs;
+    StringSchemaLocater locater;
+    for (Utf8String ec2XmlString : ec2XmlStrings)
+        {
+        SchemaKey key;
+        SchemaReadStatus status = ECSchema::ReadSchemaKey(ec2XmlString, key);
+        if (SchemaReadStatus::Success != status)
+            return BE_SQLITE_ERROR;
+        locater.AddSchemaString(key.GetName(), ec2XmlString);
+        schemaKeyPairs.push_back(std::make_pair(key, nullptr));
+        }
+    schemaContext->AddSchemaLocater(locater);
+
+    for (int i = 0; i < ec2XmlStrings.size(); i++)
+        {
+        Utf8String ec2XmlString = ec2XmlStrings[i];
+        ECSchemaPtr schema;
+        SchemaReadStatus schemaStatus = ECSchema::ReadFromXmlString(schema, ec2XmlString.c_str(), *schemaContext);
+
+        if (SchemaReadStatus::DuplicateSchema == schemaStatus && schemaKeyPairs[i].second == nullptr)
+        {
+            schema = ECSchema::LocateSchema(schemaKeyPairs[i].first, *schemaContext);
+            schemaKeyPairs[i].second = schema;
+            continue;
+        }
+
+        if (SchemaReadStatus::Success != schemaStatus)
+            return BE_SQLITE_ERROR;
+
+        schemaKeyPairs[i].second = schema;
+        }
+
+    if (0 == schemaKeyPairs.size())
+        return BE_SQLITE_ERROR;
+
+    for (bpair<SchemaKey, ECSchemaPtr>& schemaPair : schemaKeyPairs)

```

Comment:
I think you need to order by dependency order before you convert.  That way base property definitions from a referenced schema will be converted before the derived ones

Reply:
Added code to sort the schemas in dependency order prior to conversion